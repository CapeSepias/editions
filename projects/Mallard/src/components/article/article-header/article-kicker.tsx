import React from 'react'
import { View } from "react-native";
import { useArticleAppearance } from 'src/theme/appearance'
import { HeadlineKickerText } from 'src/components/styled-text';
import { longReadHeaderStyles, newsHeaderStyles } from './article-header-styles';

export interface ArticleKickerProps {
    kicker: string,
    type?: 'news' | 'longRead'
};

export const ArticleKicker = ({
    kicker,
    type
}: ArticleKickerProps) => {
    const { appearance } = useArticleAppearance();
    return (
        <View style={[type && type === 'longRead' ? longReadHeaderStyles.kicker : newsHeaderStyles.kicker, appearance.backgrounds]}>
            <HeadlineKickerText style={[appearance.text]    }>
                {kicker}
            </HeadlineKickerText>
        </View>
    );
}
