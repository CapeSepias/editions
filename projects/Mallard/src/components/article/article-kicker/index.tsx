import React from 'react'
import { View } from 'react-native'
import { useArticleAppearance } from 'src/theme/appearance'
import { HeadlineKickerText } from 'src/components/styled-text'
import { longReadHeaderStyles, newsHeaderStyles } from '../styles'

export interface ArticleKickerProps {
    kicker: string
    type?: 'news' | 'longRead'
}

export const ArticleKicker = ({ kicker, type }: ArticleKickerProps) => {
    const { appearance } = useArticleAppearance()
    return (
        <View
            style={[
                type && type === 'news'
                    ? newsHeaderStyles.kicker
                    : longReadHeaderStyles.kicker,
                appearance.backgrounds,
            ]}
        >
            <HeadlineKickerText style={[appearance.text]}>
                {kicker}
            </HeadlineKickerText>
        </View>
    )
}
