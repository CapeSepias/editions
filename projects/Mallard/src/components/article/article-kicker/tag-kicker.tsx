import React from 'react'
import { View, StyleSheet, StyleProp, ViewStyle } from 'react-native'
import { HeadlineKickerText } from 'src/components/styled-text'
import { color } from 'src/theme/color'
import { metrics } from 'src/theme/spacing'
import { getFont } from 'src/theme/typography'
import { useArticle } from 'src/hooks/use-article'
import { useMediaQuery } from 'src/hooks/use-screen'
import { Breakpoints } from 'src/theme/breakpoints'

const height = metrics.vertical * 4

const styles = StyleSheet.create({
    kicker: {
        color: color.palette.neutral[100],
        padding: metrics.article.sides,
        paddingVertical: metrics.vertical / 2,
        height,
        marginTop: metrics.vertical * -4,
        width: 'auto',
        textAlign: 'left',
        flexShrink: 1,
        fontFamily: getFont('headline', 1, 'bold').fontFamily,
    },
    kickerHolder: {
        flexDirection: 'column',
        alignItems: 'flex-start',
    },
})

interface PropTypes {
    children: string
    style?: StyleProp<ViewStyle>
}

const TagKicker = ({ children, style }: PropTypes) => {
    const [articleColor] = useArticle()
    return (
        <View style={styles.kickerHolder}>
            <HeadlineKickerText
                style={[
                    styles.kicker,
                    style,
                    {
                        backgroundColor: articleColor.main,
                    },
                ]}
            >
                {children}
            </HeadlineKickerText>
        </View>
    )
}
TagKicker.height = height

const hangyKickerStyles = StyleSheet.create({
    normal: {
        marginTop: -height,
    },
    mobile: {
        marginLeft: metrics.article.sides * -1,
    },
})
const HangyTagKicker = ({
    translate,
    ...props
}: PropTypes & { translate?: number }) => {
    const isTablet = useMediaQuery(width => width >= Breakpoints.tabletVertical)
    return (
        <TagKicker
            {...props}
            style={[
                props.style,
                hangyKickerStyles.normal,
                !isTablet && hangyKickerStyles.mobile,
                translate
                    ? {
                          transform: [{ translateY: translate }],
                      }
                    : null,
            ]}
        />
    )
}

export { TagKicker, HangyTagKicker }
