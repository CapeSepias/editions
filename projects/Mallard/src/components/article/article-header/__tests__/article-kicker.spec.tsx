import React from "react";
import TestRenderer from "react-test-renderer";
import { ArticleKicker } from "../article-kicker";

describe("ArticleKicker", () => {
    it('should display a default Kicker', () => {
        const component: any = TestRenderer.create(<ArticleKicker kicker="Opinion" />).toJSON();
        expect(component).toMatchSnapshot();
    })
    it('should display a long read style kicker', () => {
        const component: any = TestRenderer.create(<ArticleKicker kicker="Opinion" type="longRead" />).toJSON();
        expect(component).toMatchSnapshot();
    })
});
