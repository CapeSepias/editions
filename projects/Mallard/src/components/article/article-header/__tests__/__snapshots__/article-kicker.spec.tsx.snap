// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleKicker should display a default Kicker 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "marginBottom": 3,
        "paddingBottom": 6,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#dcdcdc",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 19,
          "lineHeight": 22,
          "marginBottom": 1.5,
        },
        Array [
          Object {
            "color": "#121212",
          },
        ],
      ]
    }
  >
    Opinion
  </Text>
</View>
`;

exports[`ArticleKicker should display a long read style kicker 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "marginBottom": 3,
        "paddingBottom": 6,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#dcdcdc",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 19,
          "lineHeight": 22,
          "marginBottom": 1.5,
        },
        Array [
          Object {
            "color": "#121212",
          },
        ],
      ]
    }
  >
    Opinion
  </Text>
</View>
`;
